### PR TITLE
Simplify `AbstractBuiltInCodeStyleDiagnosticAnalyzer` constructors

### PR DESCRIPTION
**Pull request by Youssef1313**
Replaces https://github.com/dotnet/roslyn/pull/62736

Addresses third point of https://github.com/dotnet/roslyn/issues/62666, and is follow-up to https://github.com/dotnet/roslyn/pull/62766 to remove unused `language` parameters

### CODE REVIEW

**Comment by Youssef1313** [comment]
@mavasani Can you take a look? Thanks!

File: src/Analyzers/CSharp/Analyzers/UseIsNullCheck/CSharpUseIsNullCheckForCastAndEqualityOperatorDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public CSharpUseIsNullCheckForCastAndEqualityOperatorDiagnosticAnalyzer()
             : base(IDEDiagnosticIds.UseIsNullCheckDiagnosticId,
                    EnforceOnBuildValues.UseIsNullCheck,
                    CodeStyleOptions2.PreferIsNullCheckOverReferenceEqualityMethod,
-                   CSharpAnalyzersResources.Use_is_null_check,
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.Use_is_null_check), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)),
```
Comment (mavasani) [inline src/Analyzers/CSharp/Analyzers/UseIsNullCheck/CSharpUseIsNullCheckForCastAndEqualityOperatorDiagnosticAnalyzer.cs:27]: Bug fix?

Comment (Youssef1313) [reply src/Analyzers/CSharp/Analyzers/UseIsNullCheck/CSharpUseIsNullCheckForCastAndEqualityOperatorDiagnosticAnalyzer.cs:27]: Yes I expect the original code has a localization problem.

I can separate it in a different PR if you prefer.

Comment (mavasani) [reply src/Analyzers/CSharp/Analyzers/UseIsNullCheck/CSharpUseIsNullCheckForCastAndEqualityOperatorDiagnosticAnalyzer.cs:27]: Keeping in this PR is fine, thanks.

**Comment by Youssef1313** [comment]
@mavasani This is ready for review.

**Review by mavasani — APPROVED** [review]

**Pull request merged**